### PR TITLE
[UXE-5176] feat: migrate edge application list to V4 in a new lazy component

### PR DESCRIPTION
**Pull request by peterpaulodev**
### What is the new behavior introduced by this PR?
- Lazy component for on-demand data retrieval.
- Upgraded list service to use the V4 API.

### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.

https://github.com/user-attachments/assets/03981c21-e20b-40bd-96b8-4a0ae1665f38

image

### Does it have a link on Figma?
no

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [x] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [x] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: src/services/edge-application-device-groups-services/v4/list-device-groups-service.js
Comment (aloisio-m-bastian) [inline src/services/edge-application-device-groups-services/v4/list-device-groups-service.js:1]: quanto aos testes, chegou a dar uma olhada se da para reutilizar os testes unitarios dos serviços da api v3?

**Review by aloisio-m-bastian — DISMISSED** [review]

File: src/services/edge-application-device-groups-services/v4/list-device-groups-service.js
Comment (peterpaulo-azion) [reply src/services/edge-application-device-groups-services/v4/list-device-groups-service.js:1]: outros foram criados, mas deu para se basear neles.

**Review by HerbertJulio — APPROVED** [review]

**Review by HerbertJulio — APPROVED** [review]

File: cypress/e2mock/edge-application/list-edge-applications.cy.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    cy.get(selectors.list.searchInput).clear()
+    cy.get(selectors.list.searchInput).type('foo 1{enter}')
+
+    cy.wait('@filteredEdgeApplicationsListApi')
+      .its('response.body')
+      .should('have.property', 'count', 1)
+    cy.get(selectors.list.filteredRow.column('name')).should('have.text', 'foo 1')
+  })
+
+  it('should ordering edge applications list', function () {
```
Comment (vinigfer) [inline cypress/e2mock/edge-application/list-edge-applications.cy.js:33]: Minor typo
```suggestion
  it('should order edge applications list', function () {
```

File: cypress/e2mock/edge-application/list-edge-applications.cy.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        req.reply({
+          fixture: '/edge-application/edge-applications-ordering-name.json'
+        })
+      }
+    }).as('filteredEdgeApplicationsListApi')
+
+    cy.get(selectors.list.orderingHeader.firstColumn).click()
+
+    cy.wait('@filteredEdgeApplicationsListApi').then((interception) => {
+      expect(interception.response.body.results[0].name).to.eq('foo 4')
```
Comment (vinigfer) [inline cypress/e2mock/edge-application/list-edge-applications.cy.js:45]: This sounds a bit weird ...
We are ordering by name (ASC) right? So shouldn't we be expecting `foo 1` over here?

**Review by vinigfer — APPROVED** [review]

**Pull request merged**